### PR TITLE
Update affiliate links and add some integration tests

### PR DESCRIPTION
**Pull request by bfalling**
Apologies for the clutter of the automatic whitespace cleanup!

### CODE REVIEW

File: openlibrary/macros/AffiliateLinks.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
+$def with (opts)
+
+$ isbn = opts['isbn']
+$ asin = opts['asin']
+$ scribd = opts['scribd']
```
Comment (mekarpeles) [inline openlibrary/macros/AffiliateLinks.html]: Is the `opts` which gets passed into the templator always guaranteed to have these three keys: isbn, asin, scribd? What happens if the key is not present? Would it make sense to do something like opts.get('isbn', '')?

File: openlibrary/macros/databarWork.html [hunk 2]
```diff
@@ -63,10 +53,10 @@
 
 $ library = None
 <style type="text/css">
-#edit-identifiers {
```
Comment (mekarpeles) [inline openlibrary/macros/databarWork.html]: Consider isolating whitespace changes on their own branch or commit (so it's easier to detect changes of substance).

File: openlibrary/macros/AffiliateLinks.html [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+$ abebooks = "http://affiliates.abebooks.com/c/64277/77798/2029?u=http%3A%2F%2Fwww.abebooks.com%2Fservlet%2FSearchResults%3Fisbn%3DXXX"
+
+$ bibliocom = "http://www.biblio.com/editions.php?isbn=XXX&aid=openlibrary"
+
+$ bookdepository = "https://www.bookdepository.com/search?searchIsbn=XXX"
+
+$ powells = "http://www.powells.com/partner/44336/biblio/XXX"
+
+<ul>
+$if isbn or asin:
```
Comment (mekarpeles) [inline openlibrary/macros/AffiliateLinks.html]: Alternatively you could replace the initial check of `$if isbn or asin` by instead changing the final elif to: `if scribd and not (asin or isbn)`. This seems more logical as it reflects the constraint that whether to show scribd depends on the presence of isbn or asin -- not the isbn and asin blocks.

File: openlibrary/macros/databarWork.html [hunk 4]
```diff
@@ -63,10 +53,10 @@
 
 $ library = None
 <style type="text/css">
-#edit-identifiers { 
+#edit-identifiers {
                     font-size: 75%;
```
Comment (mekarpeles) [inline openlibrary/macros/databarWork.html]: (a) is there a reason the css indentation is inconsistent, and (b) is there a reason these styles are not imported from a stylesheet?

File: openlibrary/templates/history/sources.html [hunk 5]
```diff
@@ -20,22 +20,22 @@
...
         url = "/show-records/" + record_id
-        
+
         if item.startswith("ia:"):
             source_name = "Internet Archive"
             source_url = "//archive.org/details/" + item[3:]
         elif item.startswith("amazon:"):
             source_name = "amazon.com"
-            source_url = "http://www.amazon.com/?tag=openlibr-20"
+            source_url = "http://www.amazon.com/?tag=internetarchi-20"
```
Comment (mekarpeles) [inline openlibrary/templates/history/sources.html]: It seems like these should be configurable in a central location to DRY them out (and passed in through the template -- for the exact case we see here).

We may want to check that we're updating this consistently across all files:
https://github.com/internetarchive/openlibrary/search?utf8=%E2%9C%93&q=openlibr-20

**Comment by mekarpeles** [comment]
Overall, the changes look good. In terms of possible regressions, we may want to DRY these affiliate links/codes as right now it seems possible for files to reference links inconsistently.

It seems like we're leveraging the templator quite a bit (it's my personal preference to do as much work within the model and python view as we can as it's easier to document (and generate documentation for), but that's just my experience.

Otherwise, LGTM

File: openlibrary/macros/databarWork.html (hunk 4 above)
Comment (bfalling) [inline openlibrary/macros/databarWork.html]: No reason for either of these. Existing CSS needs refactoring at a later time.

**Comment by bfalling** [comment]
All issues above addressed, except for the existing CSS indentation and lack of use of import.

The PR branch has been force pushed with the updates.

File: tests/integration/test_affiliate_links.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    @pytest.fixture
+    def browser(self):
+        browser = Browser('chrome')
+        yield browser
+        browser.quit()
+
+    def test_affiliate_links_on_work(self, browser):
+        url = self.host + '/works/OL6037022W/Remix'
+        browser.visit(url)
+        assert browser.is_element_present_by_css("a[href*='www.amazon.com/dp/1408113473/?tag=internetarchi-20']")
```
Comment (mekarpeles) [inline tests/integration/test_affiliate_links.py]: Again, we may want to be careful about DRY affiliate tags (e.g. these tests succeeding if/after they change)

File: openlibrary/macros/AffiliateLinks.html [hunk 7]
```diff
@@ -6,7 +6,7 @@
 
 $ alibris = "http://click.linksynergy.com/fs-bin/click?id=ECh45gHEoh4&offerid=99238.122856000&type=2&subid=0&tmpid=939&RD_PARM1=http%253A%252F%252Fwww.alibris.com/booksearch%253Fqsort%253Dp%2526isbn%253DXXX"
 
-$ amazon = "http://www.amazon.com/dp/XXX/?tag=internetarchi-20"
+$ amazon = "http://www.amazon.com/dp/XXX/?tag=" + affiliate_id('amazon')
```
Comment (mekarpeles) [inline openlibrary/macros/AffiliateLinks.html]: This seems like a big improvement -- we may want to utilize the same technique for our integration test(s)

**Pull request merged**